Reorganization of files (no source files)

zarov opened the pull request:
Little proposition for a reorganization of files, as I find it really messy as it is now:
- move all jsdoc files to a `docs/` folder
- move Github templates in  `.github/` folder
- remove unused `TESTING.md` and `CODING.md`
- move content from `BUILDING.md` inside `README.md`

zarov committed:
chore(docs): reorganize documentation files in docs/

zarov committed:
chore: move githubs templates in .github/ folder

tbroyer commented on CODING.md:
```diff
@@ -1,3 +0,0 @@
-# iTowns coding guide
-
-This guide gives advices and conventions on how to keep your code clean and coherent with the iTowns codebase.
```
Ideally, we'd want this to be expanded rather than removed.

I wouldn't mind if we chose to merge CODING and TESTING into CONTRIBUTING (i.e., removing CODING and TESTING, but keeping CONTRIBUTING and deciding that's where those sections would go if we'd want to add them in the future), but CONTRIBUTING is quite big already; so maybe instead merge BUILDING and TESTING into CODING.

tbroyer commented on README.md:
```diff
@@ -65,31 +65,57 @@ See our [release page](https://github.com/iTowns/itowns/releases).
+   * You can then require `src/Main.js` in your code
+* Test and lint changes: see test, lint and test-examples npm script
+
+### Tests
```
I'd rather move the Tests section to `BUILDING.md` (or merge everything into CODING) than mixing "what is iTowns and how to use it" and "how to build it and contribute" into a big README.

zarov replied:
Sounds good to me

tbroyer commented on package.json:
```diff
@@ -5,7 +5,7 @@
   "main": "lib/Main.js",
   "scripts": {
     "lint": "eslint \"src/**/*.js\" \"test/**/*.js\" \"examples/**/*.js\"",
-    "doc": "jsdoc -t node_modules/docdash --readme JSDOC.md -c jsdoc-config.json",
```
If this no longer uses `docdash`, should it be removed as a dependency?

Also, mention of docdash in `layout.tmpl` should probably be removed.

zarov replied:
Oh yes completely forgot about that !

zarov replied:
In fact no, `docdash`  is still mentioned in `docs/config.json`

tbroyer commented on docs/templates/layout.tmpl:
```diff
@@ -9,10 +9,10 @@
     <!--[if lt IE 9]>
       <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
```
This no longer works, for quite a long time now, maybe time to remove this as we're at it (and stop pretending IE8 is still a thing :wink:)

zarov replied:
Yep !

tbroyer reviewed (commented):
:+1: to moving GitHub templates to `.github`
:+1: to moving jsdoc-specific files into a subfolder (`docs` sounds good, could be `doc` too)
:-1: to merging BUILDING into README (adding a link to README point to CONTRIBUTING might be a good idea though)

tbroyer commented on CODING.md:
```diff
@@ -1,3 +1,49 @@
-This guide gives advices and conventions on how to keep your code clean and coherent with the iTowns codebase.
+iTowns needs `nodejs` and `npm` (at least 6.x) installed on the system. See the [node website](https://nodejs.org) to install node and npm.
+
+### Building
```
Should this be `##`?

tbroyer commented on CODING.md:
```diff
@@ -1,3 +1,49 @@
+
+### Tests
+
+If you want to run tests you'll need to install [puppeteer](https://github.com/GoogleChrome/puppeteer).
```
Rephrase to `In order to run the tests`? (could be done separately, we're only just moving things around here)

zarov replied:
I quickly rewrite some of this content, but it may need indeed a thorough rewrite

tbroyer reviewed (commented):
I approve this change, but let's get feedback from others here before merging.

tbroyer commented on CODING.md:
```diff
@@ -1,3 +1,61 @@
+
+## Testing
+
+In order to runt the tests,
```
runt? :wink: 

As said earlier, might be better to only move things around here, and update content in a separate commit (and/or PR), as it would make the changes "visible" by diffing.

zarov replied:
As you like ! I'm reverting back to the first states then

zarov committed:
chore: reorganize contents in README and other markdown files

Test section of README goes into CODING, and contents of BUILDING and
TESTING go into CODING as well.
